lte/gateway/fabfile.py

```diff
@@ -59,147 +54,172 @@
-            if vm and vm.startswith('magma_'):
-                mirrored_packages_file += vm[5:]
-
-            run(
-                'cat {}'.format(mirrored_packages_file)
-                + ' | xargs -I% sudo aptitude download -q2 %',
-            )
-            run('cp *.deb ~/magma-packages')
-            run('sudo rm -f *.deb')
+                c_agw.run(
+                    f'cat {mirrored_packages_file}'
+                    + ' | xargs -I% sudo aptitude download -q2 %',
+                )
+                c_agw.run('cp *.deb ~/magma-packages')
+                c_agw.run('sudo rm -f *.deb')
 
-        if all_deps:
-            pkg.download_all_pkgs()
-            run('cp /var/cache/apt/archives/*.deb ~/magma-packages')
+            if all_deps:
+                pkg.download_all_pkgs(c_agw)
+                c_agw.run('cp /var/cache/apt/archives/*.deb ~/magma-packages')
 
-        # Copy out C executables into magma-packages as well
-        _copy_out_c_execs_in_magma_vm()
+            # Copy out C executables into magma-packages as well
+            _copy_out_c_execs_in_magma_vm(c_agw)
 
 
-def openvswitch(destroy_vm='False', destdir='~/magma-packages/'):
-    destroy_vm = strtobool(destroy_vm)
+@task
+def openvswitch(c, destroy_vm=False, destdir='~/magma-packages/'):
     # If a host list isn't specified, default to the magma vagrant vm
-    if not env.hosts:
-        vagrant_setup('magma', destroy_vm=destroy_vm)
-    run('~/magma/third_party/gtp_ovs/ovs-gtp-patches/2.15/build.sh ' + destdir)
+    host_data = vagrant_setup(c, 'magma', destroy_vm=destroy_vm)
+    with Connection(
+            host_data.get("host_string"),
+            connect_kwargs={"key_filename": host_data.get("key_filename")},
+    ) as c_agw:
+        c_agw.run('~/magma/third_party/gtp_ovs/ovs-gtp-patches/2.15/build.sh ' + destdir)
 
 
-def depclean():
+@task
+def depclean(c):
     '''Remove all generated packaged for dependencies'''
     # If a host list isn't specified, default to the magma vagrant vm
-    if not env.hosts:
-        setup_env_vagrant()
-    run('rm -rf ~/magma-deps')
+    host_data = setup_env_vagrant(c, 'magma')
+    with Connection(
+        host_data.get("host_string"),
+        connect_kwargs={"key_filename": host_data.get("key_filename")},
+    ) as c_agw:
+        c_agw.run('rm -rf ~/magma-deps')
 
 
-def upload_to_aws():
+@task
+def upload_to_aws(c):
     # If a host list isn't specified, default to the magma vagrant vm
-    if not env.hosts:
-        setup_env_vagrant()
-
-    pkg.upload_pkgs_to_aws()
-
-
-def copy_packages():
-    if not env.hosts:
-        setup_env_vagrant()
-    pkg.copy_packages()
-
-
-def s1ap_setup_cloud():
+    host_data = setup_env_vagrant(c, 'magma')
+    with Connection(
+            host_data.get("host_string"),
+            connect_kwargs={"key_filename": host_data.get("key_filename")},
+    ) as c_agw:
+        pkg.upload_pkgs_to_aws(c_agw)
+
+
+@task
+def copy_packages(c):
```

Comment:
Can `setup_env_vagrant` return the connection in the/or next to the `host_data`? Might be possible to reduce some boilerplate by that.

Reply:
Good point! I also made the use of `setup_env_vagrant` and `vagrant_setup` consistent, because they are essential equivalent.